src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,206 @@
+            }
+            System.out.println(horizontalLine);
+            System.out.println("Bye. Hope to see you again soon");
+            System.out.println(horizontalLine);
+    }
+
+    public static void dealWithException(String text){
+        if (text.equals("todo")) {
+            System.out.println("☹ OOPS!!! The description of a todo cannot be empty.");
+        } else if (text.equals("deadline")) {
+            System.out.println("☹ OOPS!!! The description of a deadline cannot be empty.");
+        } else if (text.equals("event")) {
+            System.out.println("☹ OOPS!!! The description of a event cannot be empty.");
+        } else if(text.contains("done")){
+            System.out.println("☹ OOPS!!! The index is out of bound");
+        } else if(text.contains("delete")){
+            System.out.println("☹ OOPS!!! The index is out of bound");
+        } else {
+            System.out.println(("☹ OOPS!!! I'm sorry, but I don't know what that means :-("));
+        }
+    }
+
+    public static void addTodoTask(String text) throws DukeException {
+        String todoDescription;
+        if(text.equals("todo")){
+            throw new DukeException();
+        }
+        todoDescription = text.substring(5);
+        Task task = new Todo(todoDescription);
+        //tasks[thingsCounted] = task;
+        thingsCounted++;
+        taskArray.add(task);
+        printTask(task);
+    }
+
+    public static void addDeadlineTask (String text) throws DukeException {
+        String deadlineDescription;
+        String deadlineByDate;
+        int getIndex;
+        if(text.equals("deadline")){
+            throw new DukeException();
+        }
+        getIndex = text.indexOf("/");
+        deadlineDescription = text.substring(9, getIndex - 1);
+        deadlineByDate = text.substring(getIndex + 4);
+        Task task = new Deadline(deadlineDescription, deadlineByDate);
+        //tasks[thingsCounted] = task;
+        thingsCounted++;
+        taskArray.add(task);
+        printTask(task);
+    }
+
+    public static void addEventTask(String text) throws DukeException {
+        String eventDescription;
+        String eventAtDate;
+        int getIndex;
+        if(text.equals("event")){
+            throw new DukeException();
+        }
+        getIndex = text.indexOf("/");
+        eventDescription = text.substring(6, getIndex - 1);
+        eventAtDate = text.substring(getIndex + 4);
+        Task task = new Event(eventDescription, eventAtDate);
+        //tasks[thingsCounted] = task;
+        thingsCounted++;
+        taskArray.add(task);
+        printTask(task);
+    }
+
+    public static void printTask(Task task){
+        System.out.println("Got it. I've added this task: ");
+        System.out.println(task);
+        System.out.println("Now you have " + thingsCounted + " tasks in the list.");
+    }
+
+
+    public static void markTaskAsDone(int index){
+        taskArray.get(index-1).taskDone();
```

Comment:
Do note of using of magic literals, perhaps name your `index` more descriptively so that the reader can understand why -1 is required.